src/libponyc/verify/fun.c

```diff
@@ -528,13 +528,12 @@ bool verify_behaviour_root_struct_param(pass_opt_t* opt, ast_t* type, bool is_in
       {
         ast_error(opt->check.errors, type_node,
           "cannot use a tuple with a struct type member as parameter to a behaviour; "
-          "this would be possible if the capability was \"tag\", "
-          "or if the parameter was wrapped inside a class or actor");
+          "this would be possible if the capability of the struct was \"tag\", "
+          "or if it was wrapped inside a class or actor");
       } else {
         ast_error(opt->check.errors, type_node,
-          "cannot use a struct type as parameter to a behaviour; "
-          "this would be possible if the capability was \"tag\", "
-          "or if the parameter was wrapped inside a class or actor");
+          "only \"tag\" structs can be used as parameters to a behaviour; "
+          "you will need to change the capability, or wrap the struct in a class or actor");
```

Comment:
This wording should be the same as the above.
But in any case we should put "if it was wrapped in a class or actor" first because it's not particularly likely that it will fix the users' problems whereas wrapping in a class or actor would

Reply:
> This wording should be the same as the above.

Do you mean that both should use the "this would be possible if..." wording? It's also my personal choice, but I think that the first sentence of the error should clearly explain what's wrong. In that sense, Sean's choice of "only tag structs can be used" is better, but I haven't found an easy way to word it that way in the tuple case.

> we should put "if it was wrapped in a class or actor" first 

Agreed.

Reply:
For the first point I meant that the tuple error message and the non-tuple error message are currently worded quite differently from each other. I.e. I expected you wanted to change both but changed only one

Reply:
I guess specifically what's off to me is this case has the second person:
" you will need to ..." whereas the other has the impersonal: "this would be possible if ..."

Reply:
Ah, right, seems like I forgot to change the second part of the message